index.ts

```diff
@@ -188,23 +207,31 @@ function main(contents: string): {}[] {
   // convert JsxOpeningLikeElements to Message maps
   const jsxMessages = elements
     .map((element) => {
-      const msg: Message = {};
-      element.attributes &&
+      const msg: PartialMessage = {};
+      if (element.attributes) {
         element.attributes.properties.forEach((attr: ts.JsxAttributeLike) => {
           // found nothing
           // tslint:disable-next-line:no-any
           const a = attr as any; // TODO find correct types to avoid "any"
           if (!a.name || !a.initializer) {
             return;
           }
-          msg[a.name.text] =
-            a.initializer.text || a.initializer.expression.text;
+          copyIfMessageKey(
+            msg,
+            a.name.text,
+            a.initializer.text || a.initializer.expression.text,
+          );
         });
-      return msg;
+      }
+      return isValidMessage(msg) ? msg : null;
     })
-    .filter((r) => !emptyObject(r));
+    .filter(notNull);
 
   return jsxMessages.concat(dm).concat(fm);
 }
 
+function notNull<T>(value: T | null): value is T {
```

Comment:
It does return a boolean at runtime, but this bit of typing enables the result of `filter` to be `Message` instead of `Message | null` as far as the type checker is concerned. It's a user-defined type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html